File: leet-code/Algorithms/217-contains_duplicate/sol_a.rs
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        use std::collections::HashMap;
        
        let mut numMap = HashMap::new();
        
        for n in nums {
            let count = numMap.entry(n).or_insert(0);
            *count += 1;
        }
        
        for (key, count) in numMap {
            if count > 1 {
                return true;
            }
        }
        
        false
    }
}

File: leet-code/Algorithms/026-remove_duplicates_from_sorted_array/sol_a.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        if nums == []:
            return 0
        
        count = 1
        
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[count] = nums[i]
                count += 1
            
        return count
